packages/react/src/types.ts

```diff
@@ -77,7 +77,9 @@ type ButtonProps = {
 };
 
 export type SignInButtonProps = ButtonProps;
-export type SignUpButtonProps = ButtonProps;
+export interface SignUpButtonProps extends ButtonProps {
```

Comment:
❓Any specific reason we want to switch to an interface and not go with something like this ?
```ts
export type SignUpButtonProps = ButtonProps & {unsafeMetadata?: SignUpUnsafeMetadata;}
```

Reply:
@panteliselef While not exactly this situation, I tend to refer to this: https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections

That said, happy to use an intersection.

Reply:
that's a great read, thanks tom